price-feeder/config/currency_provider_tracker.go

```diff
@@ -168,7 +192,7 @@ func (t *CurrencyProviderTracker) setCurrencyProviders() error {
 // set to 3 providers.
 func (t *CurrencyProviderTracker) setCurrencyProviderMin() {
 	for base, exchanges := range t.CurrencyProviders {
-		if len(exchanges) < 3 {
+		if len(exchanges) < 3 && len(exchanges) > 0 { // can be 0 if request errored
 			t.CurrencyProviderMin[base] = len(exchanges)
 		} else {
 			t.CurrencyProviderMin[base] = 3
```

Comment:
What if it's zero? Do we handle that case anywhere?

Reply:
I added a log if the coin gecko response actually returns 0 providers without (not from erroring out). I think it should still default to a three provider minimum in this case, which is what would happen in the current implementation.

Reply:
I'm still not sure why we cap it to 3, and how do we handle the zero case. Could you explain it please?

Reply:
Three is just the minimum providers we enforced before supporting the lower cap assets which is why I'm setting that as a default minimum if the coingecko request for the providers supporting a certain pair errors out. In the case where coin gecko just returns 0 (not from erroring out), that would suggest that coin gecko knows no providers for the supplied pair. In that scenario, it makes sense to default our three provider minimum rule since we have no way of knowing how many providers support that pair if coin gecko doesn't have data on it. This is something that would most likely happen from someone putting in a pair that actually doesn't have any providers, so the log would inform them that coin gecko isn't aware of any providers supporting that pair.

Reply:
OK, makes sense. Please add a comment in the code.